cmd/master.go

```diff
@@ -60,14 +62,121 @@ import (
 	"github.com/jackc/pgconn"
 	"go.uber.org/dig"
 	"go.uber.org/zap"
+	"golang.org/x/crypto/bcrypt"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/reflection"
 	"google.golang.org/grpc/status"
 	"gorm.io/gorm"
+
+	"github.com/spf13/cobra"
 )
 
+// masterCmd represents the master command
+var masterCmd = &cobra.Command{
+	Use:   "master",
+	Short: "master runs the grpc server and runner if in single-node mode",
+	Run: func(cmd *cobra.Command, args []string) {
+		log.Println("master called")
+
+		d, err := di.BuildMasterContainer(C)
+		if err != nil {
+			log.Panicf("%v", err)
+		}
+
+		err = SetupDB(d)
+		if err != nil {
+			log.Panicf("Error setting up the database: %v", err)
+		}
+
+		db := storage.GetGlobalDB()
+
+		err = sutil.CreateAllTables(db)
+		if err != nil {
+			log.Panicf("%v", err)
+		}
+
+		err = sutil.LoadConfig(db, &D)
+		if err != nil {
+			log.Panicf("%v", err)
+		}
+		err = sutil.LoadReport(db)
+		if err != nil {
+			log.Panicf("%v", err)
+		}
+
+		store := diutil.NewStore()
+
+		var q queue.WorkerQueue
+		di.Invoke(func(qu queue.WorkerQueue) {
+			q = qu
+		})
+
+		dr := runner.NewRunner(db, q, store, C)
+		go func() {
+			err := dr.MasterRunner()
+			if err != nil {
+				log.Panicf("%v", err)
+			}
+		}()
+
+		err = Start(C, d, db)
+		if err != nil {
+			log.Panicf("%v", err)
+		}
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(masterCmd)
+
+	// Here you will define your flags and configuration settings.
+
+	// Cobra supports Persistent Flags which will work for this command
+	// and all subcommands, e.g.:
+	// masterCmd.PersistentFlags().String("foo", "", "A help for foo")
+
+	// Cobra supports local flags which will only run when this command
+	// is called directly, e.g.:
+	// masterCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+}
+
+func SetupDB(cont *dig.Container) error {
+	var db *gorm.DB
+	err := cont.Invoke(func(d *gorm.DB) {
+		db = d
+	})
+	if err != nil {
+		return err
+	}
+	var tm time.Time
+	res, err := db.Raw("SELECT current_timestamp;").Rows()
+	if err != nil {
+		panic(err)
```

Comment:
lets not panic. this error could be somewhat expected, instead, lets return it to the parent function:

return err

Reply:
https://github.com/ScoreTrak/ScoreTrak/pull/248/commits/8bb96233d1e79eefda053e717b13329064d0513b